flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/logical/groupWindows.scala

```diff
@@ -39,164 +34,39 @@ case class LogicalOverWindow(
     following: Optional[Expression])
 
 // ------------------------------------------------------------------------------------------------
-// Tumbling group windows
+// Group windows
 // ------------------------------------------------------------------------------------------------
 
+sealed trait LogicalWindow {
+  def timeAttribute: PlannerExpression
+  def aliasAttribute: PlannerExpression
+}
+
 case class TumblingGroupWindow(
-    alias: PlannerExpression,
-    timeField: PlannerExpression,
+    aliasAttribute: PlannerExpression,
+    timeAttribute: PlannerExpression,
     size: PlannerExpression)
-  extends LogicalWindow(
-    alias,
-    timeField) {
-
-  override def resolveExpressions(
-      resolve: (PlannerExpression) => PlannerExpression): LogicalWindow =
-    TumblingGroupWindow(
-      resolve(alias),
-      resolve(timeField),
-      resolve(size))
-
-  override def validate(tableEnv: TableEnvironment): ValidationResult =
-    super.validate(tableEnv).orElse(
-      tableEnv match {
-
-        // check size
-        case _ if !isTimeIntervalLiteral(size) && !isRowCountLiteral(size) =>
-          ValidationFailure(
-            "Tumbling window expects size literal of type Interval of Milliseconds " +
```

Comment:
where can I find this check for all types of windows?

Reply:
Actually I somehow missed that check. We also didn't have tests for this case unfortunately. I reintroduced this check and added tests to validate it.